### ISSUE TITLE
generated go struct wrapped to slice cause panic

### ISSUE THREAD

**Issue by elvizlai** [goccy/go-json#151]
proto file
```
package common;

message Id {
  string id = 1;
}
```

go code
```
package main

import (
	"fmt"
        "encoding/json"

        gjson "github.com/goccy/go-json"

	"xxx/common"
)

func main() {
       // common.Id is proto generated go struct
	type Data struct {
		A []*common.Id
	}

	d := &Data{A: []*common.Id{{Id: "1"}, {Id: "2"}}}

	// fmt.Println(json.Marshal(d)) this works
    
        // this panics
	fmt.Println(gjson.Marshal(d))
}
```

This panic on both v0.4.7 or master branch
https://github.com/goccy/go-json/blob/7c0a0560647614f85336cd2daac495f719becb7a/encode_vm_escaped.go#L166

**Comment by goccy** [comment]
I tried to reproduce this issue, but I couldn't.
Please try again on the latest master branch or write the source code to reproduce in the Go playground and share the URL.

**Comment by elvizlai** [comment]
@goccy Please try this: https://github.com/elvizlai/gjson-test-case

**Comment by goccy** [comment]
Thank you for the preparation of example project .
I could reproduce this problem, and fixed with #153 !
